docs/spec/src/integration/spec/5-lifecycle-phases.md

```diff
@@ -0,0 +1,169 @@
+  bytes length_proof = 3;
+  // Length in bn254 field elements (32 bytes) of the blob. Must be a power of 2.
+  uint32 length = 4;
+}
+```
+
+Unlike Ethereum blobs which are all 128KiB, EigenDA blobs can be any power of 2 length between 32KiB and 16MiB (currently), and so the `commitment` alone is not sufficient to prevent certain attacks:
+
+- Why is a commitment to the length of the blob necessary?
+    
+    There are different variants of the attack. The basic invariant the system needs to satisfy is that with the chunks from sufficient set of validators, you can get back the full blob. So the total size of the chunks held by these validators needs to exceed the blob size. If I don't know the blob size (or at least an upper bound), there's no way for the system to validate this invariant.
+    Here’s a simple example. Assume a network of 8 DA nodes, and coding ratio 1/2. For a `blob` containing 128 field elements (FEs), each node gets 128*2/8=32 FEs, meaning that any 4 nodes can join forces and reconstruct the data. Now assume a world without length proof; a malicious disperser receives the same blob, uses the same commitment, but claims that the blob only had length 4 FEs. He sends each node 4*2/8=1 FE. The chunks submitted to the nodes match the commitment, so the nodes accept and sign over the blob’s batch. But now there are only 8 FEs in the system, which is not enough to reconstruct the original blob (need at least 128 for that).
+    
+
+> Note that the length here is the length of the blob (power of 2), which is different from the payload_length encoded as part of the `PayloadHeader` in the `blob` itself (see the [encoding section](#encoding)).
+> 
+
+**PaymentHeader**
+
+The paymentHeader specifies how the blob dispersal to the network will be paid for. There are 2 main modes of payment, the permissionless pay-per-blob model and the permissioned reserved-bandwidth approach. See the [Payments](https://docs.eigenda.xyz/releases/payments) release doc for full details; we will only describe how to set these 4 fields here.
+
+```protobuf
+message PaymentHeader {
+  // The account ID of the disperser client. This should be a hex-encoded string of the ECSDA public key
+  // corresponding to the key used by the client to sign the BlobHeader.
+  string account_id = 1;
+  uint32 reservation_period = 2;
+  bytes cumulative_payment = 3;
+    // The salt is used to ensure that the payment header is intentionally unique.
+  uint32 salt = 4;
+}
+```
+
+Users who want to pay-per-blob need to set the cumulative_payment. Users who have already paid for reserved-bandwidth can set the reservation_period.
+
+An rpc call to the Disperser’s `GetPaymentState` method can be made to query the current state of an `account_id`. A client can query for this information on startup, cache it and then update it manually when making pay-per-blob payments, by keeping track of its current `cumulative_payment` to set it correctly for subsequent dispersals.
```

Comment:
> A client can query for this information on startup, cache it and then update it manually when making pay-per-blob payments, by keeping track of its current `cumulative_payment` to set it correctly for subsequent dispersals.

This sentence is a run on, consider rephrasing:

> A client can query for this information on startup, cache it, and then update it manually when making pay-per-blob payments. In this way, it can keep track of the `cumulative_payment` and set it correctly for subsequent dispersals.